msg.c

```diff
@@ -170,11 +174,26 @@ int connect_monitor(char *devname)
 
 	addr.sun_family = PF_LOCAL;
 	strcpy(addr.sun_path, path);
-	if (connect(sfd, (struct sockaddr*)&addr, sizeof(addr)) < 0) {
-		close(sfd);
-		return -1;
-	}
 
+	/* In foreground mode, when mdadm is trying to connect to control
+	 * socket it is possible that the mdmon has not created it yet.
+	 * Give some time to mdmon to create socket.
+	 */
+	while (1) {
+		rv = connect(sfd, (struct sockaddr*)&addr, sizeof(addr));
+		if (rv < 0 && retry_count >= 10) {
```

Comment:
In fact we are waiting 2,2 second (we are waiting 11 times) :) because of `retry_count >= 10`, it should be `retry_count > 10`

Reply:
I propose reworking it to `for` better readability (`for` suggest that we are iterating over something), what do you think about it?
```c
int rv = 0;

for (retry_count = 0; retry_count < 10; retry_count ++) {
    rv = connect(sfd, (struct sockaddr*)&addr, sizeof(addr));

    if (rv < 0) {
           // I didn't check if I put correct value
           sleep_for(0, 200, false);
           continue;
    }
    break;
}

if (rv) {
     pr_err("Failed to connect to control socket. (%s!!)\n", strerror(errno));
     close(sfd);
     return -1;
}
```

Reply:
looks good. I will update the patch.